### PR TITLE
[improve](serde) support const column in serialize/deserialize function

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

support const column in serialize/deserialize function

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/agent/be_exec_version_manager.h [hunk 1]
```diff
@@ -33,6 +33,8 @@ constexpr inline int AGG_FUNCTION_NULLABLE = 5; // change some agg nullable prop
 constexpr inline int VARIANT_SERDE = 6;         // change variant serde to fix PR #38413
 constexpr inline int AGGREGATION_2_1_VERSION =
         6; // some aggregation changed the data format after this version
+constexpr inline int USE_CONST_SERDE =
+        1000; // support const column in serialize/deserialize function: PR #41175
```
Comment (zhangstar333) [inline be/src/agent/be_exec_version_manager.h:37]: test old version, should Chang to 7 after run p0 cases

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/data_types/data_type_array.cpp [hunk 2]
```diff
@@ -71,45 +72,116 @@ size_t DataTypeArray::get_number_of_dimensions() const {
...
-    return get_nested_type()->serialize(data_column.get_data(), buf, be_exec_version);
+    if (be_exec_version >= USE_CONST_SERDE) {
+        // const flag
+        bool is_const_column = is_column_const(column);
+        *reinterpret_cast<bool*>(buf) = is_const_column;
+        buf += sizeof(bool);
+
+        // row num
+        const auto row_num = column.size();
+        *reinterpret_cast<ColumnArray::Offset64*>(buf) = row_num;
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_array.cpp:111]: row_num why use `ColumnArray::Offset64*`

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/columns/column_const.cpp [hunk 3]
```diff
@@ -50,6 +50,21 @@ ColumnConst::ColumnConst(const ColumnPtr& data_, size_t s_) : data(data_), s(s_)
...
 }
 
+ColumnConst::ColumnConst(const ColumnPtr& data_, size_t s_, bool create_with_empty)
+        : data(data_), s(s_) {
+    /// Squash Const of Const.
+    while (const auto* const_data = typeid_cast<const ColumnConst*>(data.get())) {
+        data = const_data->get_data_column_ptr();
+    }
+
+    if (!(data->empty() && create_with_empty)) {
```
Comment (HappenLee) [inline be/src/vec/columns/column_const.cpp:60]: which case the data will be empty()

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**